### DISCUSSION TITLE
Feature Request: PowerStream

### DISCUSSION THREAD

**Discussion by andreasbuff** [tolwi/hassio-ecoflow-cloud#222] [Ideas]
Hello, Thank you for the Integration of PowerStream. May I ask you, if it was possible, to add a "Switch" to toggle between the modes: "prioritize Battery", while loading or"prioritize Grid".  This Switch is available in the EcoFlow App: Room -> PowerStream System -> Settings: Power-Mode: Battery Prio and Grid Prio...

With this we could automate the Loading, or use of solar-power in an Automation, based on other parameters. Thanks

**Comment by Nid01** [comment]
@andreasbuff 
since the data of the PowerStream is only available as binary data via the API used by the integration there is no direct way to implement this.
The newer API https://developer-eu.ecoflow.com/us/document/powerStreamMicroInverter makes this possible, but I can't promise when @Mark-Hicks or I are able to provide a new integration using the new API.

**Comment by Tino0309** [comment]
Es wäre super wenn das kommt. Hab gerade auch danach gesucht.
DANKE
